File: cpp/src/lib/gherkin/parse_error.cpp
#include <gherkin/parse_error.hpp>

namespace gherkin {

json
parse_error::to_json() const
{
    json j;

    j["parseError"]["source"]["uri"] = uri;
    location.to_json(j["parseError"]["source"]["location"]);
    j["parseError"]["message"] = message;

    return j;
}

}

File: dotnet/Gherkin/CucumberMessages/Types/ParseError.cs
using System.Runtime.Serialization;

namespace Gherkin.CucumberMessages.Types;

public class ParseError
{
    [DataMember(Name = "source")]
    public required SourceReference Source { get; set; }

    [DataMember(Name = "message")]
    public required string Message { get; set; }
}